accounts-db/src/accounts_index/in_mem_accounts_index.rs

```diff
@@ -1125,62 +1139,83 @@ impl<T: IndexValue, U: DiskIndexValue + From<T> + Into<T>> InMemAccountsIndex<T,
             let mut flush_stats = DiskFlushStats::new();
             // we don't care about lock time in this metric - bg threads can wait
             let flush_update_measure = Measure::start("flush_update");
-            let evictions_age = evictions_age_possible
-                .into_iter()
-                .filter_map(|(k, v)| {
-                    // consider whether to write to disk for all the items we may evict
-                    let mut mse = Measure::start("flush_should_evict");
-                    let should_evict = self.should_evict_from_mem(
-                        current_age,
-                        &v,
-                        startup,
-                        true,
-                        ages_flushing_now,
-                    );
-                    mse.stop();
-                    flush_stats.flush_should_evict_us += mse.as_us();
-                    if !should_evict {
-                        // not evicting, so don't write, even if dirty
-                        return None;
-                    };
-
-                    // if we are evicting it, then we need to update disk if we're dirty
-                    if v.clear_dirty() {
-                        // step 1: clear the dirty flag
-                        // step 2: perform the update on disk based on the fields in the entry
-                        // If a parallel operation dirties the item again - even while this flush is occurring,
-                        //  the last thing the writer will do, after updating contents, is set_dirty(true)
-                        //  That prevents dropping an item from cache before disk is updated to latest in mem.
-                        // It is possible that the item in the cache is marked as dirty while these updates are happening. That is ok.
-                        //  The dirty will be picked up and the item will be prevented from being evicted.
 
+            // Process each eviction candidate
+            // For dirty entries: lock map briefly, get entry, calculate disk value, release lock, then write to disk
+            // For non-dirty entries: skip checks and pass to evict_from_cache
+            let evictions_age: Vec<_> = evictions_age_possible
+                .iter()
+                .filter_map(|(key, is_dirty)| {
+                    if *is_dirty {
+                        // Entry was dirty at scan time, need to write to disk
+                        // Extract disk data while holding map read lock, then write outside the lock
+                        let lock_measure = Measure::start("flush_read_lock");
```

Comment:
We're creating this Measure for every item in `evictions_age_possible`. Should we though? Do we think that's the bottleneck here?

When looking at perf stuff for startup in the past, we've seen that `Instant::now()` (which is in Measure), was an issue sometimes.

So seeing a measure for every iteration seems like something we should double check.